docs/spec/staking/spec-technical.md

```diff
@@ -375,111 +389,262 @@ type TxUnbond struct {
+        gs.BondedPool -= removedTokens
+        returnedCoins = exchangeRate(gs.BondedShares, gs.BondedPool) * removedTokens
+	else 
+		gs.UnbondedShares -= globalPoolSharesToRemove
+		removedTokens = exchangeRate(gs.UnbondedShares, gs.UnbondedPool) * globalPoolSharesToRemove
+        gs.UnbondedPool -= removedTokens
+        returnedCoins = exchangeRate(gs.UnbondedShares, gs.UnbondedPool) * removedTokens
+	
+	candidate.GlobalStakeShares -= removedTokens
+	candidate.IssuedDelegatorShares -= shares
+	return returnedCoins
+
+delegatorShareExRate(candidate Candidate):
+	if candidate.IssuedDelegatorShares.IsZero() then
+		return rational.One
+	
+	return candidate.GlobalStakeShares / candidate.IssuedDelegatorShares
+	
+bondedToUnbondedPool(candidate Candidate):
+
+	// replace bonded shares with unbonded shares
+	removedTokens = exchangeRate(gs.BondedShares, gs.BondedPool) * candidate.GlobalStakeShares 
+    gs.BondedShares -= candidate.GlobalStakeShares
+    gs.BondedPool -= removedTokens
 	
-removeDelegatorBond(candidate Candidate):
-
-	// first remove from the list of bonds
-	pks = loadDelegatorCandidates(store, sender)
-	for i, pk := range pks {
-		if candidate.Equals(pk) {
-			pks = append(pks[:i], pks[i+1:]...)
-		}
-	}
-	b := wire.BinaryBytes(pks)
-	store.Set(GetDelegatorBondsKey(delegator), b)
-
-	// now remove the actual bond
-	store.Remove(GetDelegatorBondKey(delegator, candidate))
-	//updateDelegatorBonds(store, delegator)
-}	
+	gs.UnbondedPool += removedTokens
+	issuedShares = removedTokens / exchangeRate(gs.UnbondedShares, gs.UnbondedPool)
+    gs.UnbondedShares += issuedShares
+    
+	candidate.GlobalStakeShares = issuedShares
+	candidate.Status = Unbonded
+
+	return transfer(address of the bonded pool, address of the unbonded pool, removedTokens)
 ```
 
-### Inflation provisions
+### TxRedelegate
+
+The re-delegation command allows delegators to switch validators while still
+receiving equal reward to as if they had never unbonded.
+
+``` golang
+type TxRedelegate struct { 
+	PubKeyFrom crypto.PubKey
+	PubKeyTo   crypto.PubKey
+	Shares     rational.Rat 
+}
+```
 
-Validator provisions are minted on an hourly basis (the first block of a new
-hour).  The annual target of between 7% and 20%. The long-term target ratio of
-bonded tokens to unbonded tokens is 67%.  
+```
+redelegate(tx TxRedelegate):
+    bond = loadDelegatorBond(store, sender, tx.PubKey)
+    if bond == nil then return 
     
-The target annual inflation rate is recalculated for each previsions cycle. The
-inflation is also subject to a rate change (positive of negative) depending or
-the distance from the desired ratio (67%). The maximum rate change possible is
-defined to be 13% per year, however the annual inflation is capped as between
-7% and 20%.
+    if bond.Shares < tx.Shares return 
+    //TODO: How we can enforce that the same bond can't be redelegated? Should we split the current bond into two so 
+    // we have a special bond for redelegation? 	
```

Comment:
When you are redegating, the shares stay with the original validator until the queue has completed... this way it is impossible to have multiple re-delegations... resolves your question?